src/parse.c

```diff
@@ -4440,7 +4440,7 @@ void Parser::checkCstyleTypeSyntax(Loc loc, Type *t, int alt, Identifier *ident)
 
 Statement *Parser::parseStatement(int flags, const utf8_t** endPtr, Loc *pEndloc)
 {
-    Statement *s;
+    Statement *s = NULL;
```

Comment:
I tried the goto, it works, but it now fails the test diag11198:

```
Test failed.  The logged output:
../src/dmd -m64 -Ifail_compilation -verrors=0   -odtest_results/fail_compilation -oftest_results/fail_compilation/diag11198_0.o -c fail_compilation/diag11198.d
fail_compilation/diag11198.d(11): Error: version conditions can only be declared at module scope


==============================
Test failed: 
expected:
----
fail_compilation/diag11198.d(11): Error: version conditions can only be declared at module scope
fail_compilation/diag11198.d(12): Error: debug conditions can only be declared at module scope
----
actual:
----
fail_compilation/diag11198.d(11): Error: version conditions can only be declared at module scope
----


make[1]: *** [test_results/fail_compilation/diag11198.d.out] Error 1
```

Is this what you would expect to happen?